tests/system-tests/rdscore/internal/rdscorecommon/log-forwarding.go

```diff
@@ -0,0 +1,255 @@
+			PodUID       []string `json:"authentication.kubernetes.io/pod-uid,omitempty"`
+		} `json:"extra,omitempty"`
+		Groups   []string `json:"groups,omitempty"`
+		UID      string   `json:"uid,omitempty"`
+		UserName string   `json:"userName,omitempty"`
+	} `json:"user,omitempty"`
+	UserAgent string `json:"userAgent,omitempty"`
+	Verb      string `json:"verb,omitempty"`
+}
+
+// VerifyLogForwardingToKafka Verify cluster log forwarding to the Kafka aggregator.
+//
+//nolint:funlen
+func VerifyLogForwardingToKafka() {
+	By("Insure CLO deployed")
+
+	var ctx SpecContext
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Verify CLO namespace %s defined", rdscoreparams.CLONamespace)
+
+	err := apiobjectshelper.VerifyNamespaceExists(APIClient, rdscoreparams.CLONamespace, time.Second)
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to pull namespace %q; %v",
+		rdscoreparams.CLONamespace, err))
+
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Verify CLO deployment %s defined in namespace %s",
+		rdscoreparams.CLODeploymentName, rdscoreparams.CLONamespace)
+
+	err = apiobjectshelper.VerifyOperatorDeployment(APIClient,
+		rdscoreparams.CLOName,
+		rdscoreparams.CLODeploymentName,
+		rdscoreparams.CLONamespace,
+		time.Minute)
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("operator deployment %s failure in the namespace %s; %v",
+			rdscoreparams.CLOName, rdscoreparams.CLONamespace, err))
+
+	By("Retrieve kafka server URL")
+	glog.V(rdscoreparams.RDSCoreLogLevel).Infof("Retrieve Kafka server URL from the ClusterLogForwarder")
+
+	clusterLogForwarder, err := clusterlogging.PullClusterLogForwarder(
+		APIClient, rdscoreparams.CLOInstanceName, rdscoreparams.CLONamespace)
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf(
+		"Failed to retrieve ClusterLogForwarder %s from the namespace %q; %v",
+		rdscoreparams.CLOInstanceName, rdscoreparams.CLONamespace, err))
+
+	clfOutput := clusterLogForwarder.Object.Spec.Outputs
+	Expect(len(clfOutput)).ToNot(Equal(0), fmt.Sprintf(
+		"No collector defined in the ClusterLogForwarder %s from the namespace %q",
+		rdscoreparams.CLOInstanceName, rdscoreparams.CLONamespace))
+
+	var kafkaURL, kafkaUser string
+
+	for _, collector := range clfOutput {
+		if collector.Type == "kafka" {
+			clfKafkaURL := collector.Kafka.URL
+
+			glog.V(100).Infof("collector.URL: %s", clfKafkaURL)
+
+			kafkaURL = strings.Split(clfKafkaURL, "/")[2]
+			kafkaUser = strings.Split(clfKafkaURL, "/")[3]
+		}
+	}
+
+	By("Getting cluster domain")
+
+	clusterDNS, err := dns.Pull(APIClient)
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf(
+		"Failed to retrieve clusterDNS object cluster from the namespace default; %v", err))
+
+	clusterDomain := clusterDNS.Object.Spec.BaseDomain
+	glog.V(100).Infof("DEBUG: clusterDomain: %s", clusterDomain)
+
+	By("Build query request command")
+
+	cmdToRun := []string{"/bin/sh", "-c", fmt.Sprintf("kcat -b %s -C -t %s -C -q -o end -c %d | grep %s",
+		kafkaURL, kafkaUser, logMessageCnt, clusterDomain)}
```

Comment:
Could U please add an option to filter by the user provided label and fall back to the cluster domain if it's not specified, thanks

Reply:
probably I need to break it into the smaller sub-methods. give me 10 mins...

Reply:
the empty clusterDomain verification added

Reply:
clusterDomain is the best way to filter-out all relevant records. There are too much, from the different servers. if we'll filter by something else it can return not correct information

Reply:
CLO's configuration allows to set custom labels on the record so it could be more precise for filtering than looking for any record that has some domain